<?xml version="1.0"?>

<launch>

    <arg name="hz" default="20"/>
    <arg name="excess_detection_ratio" default="1.2"/>
    <arg name="goal_radius" default="0.5"/>
    <arg name="enable_requesting_replanning" default="false"/>
    <arg name="intersection_acceptance_progress_ratio" default="0.7"/>
    <arg name="robot_frame" default="base_link"/>
    <arg name="goal_dis" default="5.0"/>
    <arg name="global_path_index_offset" default="1"/>
    <arg name="local_map" default="/local_map"/>
    <arg name="local_goal_angle" default="2.0"/>
    <arg name="d_local_goal_angle" default="0.4"/>
    <arg name="excess_detection_distance" default="5.0"/>
    <arg name="timeout" default="170.0"/>

    <node pkg="node_edge_navigator" type="node_edge_navigator" name="node_edge_navigator">
        <param name="HZ" value="$(arg hz)"/>
        <param name="EXCESS_DETECTION_RATIO" value="$(arg excess_detection_ratio)"/>
        <param name="GOAL_RADIUS" value="$(arg goal_radius)"/>
        <param name="ENABLE_REQUESTING_REPLANNING" value="$(arg enable_requesting_replanning)"/>
        <param name="INTERSECTION_ACCEPTANCE_PROGRESS_RATIO" value="$(arg intersection_acceptance_progress_ratio)"/>
        <param name="ROBOT_FRAME" value="$(arg robot_frame)"/>
        <param name="GOAL_DISTANCE" value="$(arg goal_dis)"/>
        <param name="GLOBAL_PATH_INDEX_OFFSET" value="$(arg global_path_index_offset)"/>
        <param name="EXCESS_DETECTION_DISTANCE" value="$(arg excess_detection_distance)"/>
        <param name="TIMEOUT" value="$(arg timeout)"/>
    </node>

    <node pkg="node_edge_navigator" type="local_goal_creator" name="local_goal_creator">
        <param name="LOCAL_GOAL_ANGLE" value="$(arg local_goal_angle)"/>
        <param name="D_LOCAL_GOAL_ANGLE" value="$(arg d_local_goal_angle)"/>
        <param name="GOAL_DIS" value="$(arg goal_dis)"/>
        <remap from="/local_map" to="$(arg local_map)"/>
        <remap from="/local_map" to="/local_map/expand"/>
    </node>

</launch>
